pkg/core/native/interop.go

```diff
@@ -38,9 +40,14 @@ func Call(ic *interop.Context) error {
 	if !ok {
 		return fmt.Errorf("method not found")
 	}
-	if !ic.VM.Context().GetCallFlags().Has(m.RequiredFlags) {
+	reqFlags := m.RequiredFlags
+	if !ic.IsHardforkEnabled(config.HFAspidochelone) && c.Metadata().ID == ManagementContractID &&
+		(m.MD.Name == "deploy" || m.MD.Name == "update") {
```

Comment:
If we're doing this for both `deploy` and `update`, then maybe it's worth to add `deployWithData` and `updateWithData` also to preserve full consistency with the old behaviour? Even if `*WithData` are not related to this state diff.

Reply:
It's `MD.Name ==`, so it works for any number of deploy/update parameters.

Reply:
Oh, somehow I've looked at the method implementation's names, not at the method's names, sorry.